### PR TITLE
In this branch I create the Meal recipes provider

### PR DESCRIPTION
**Pull request by MohammedElseddik**
## we chose the MEALDB API to create a website that provides A variety of meals and the recipes.

- Live Site https://mohammedelseddik.github.io/JavaScript-Capstone-Meals-website/
- the user can like the recipes.
- for each like, the 'likes' under the heart icon are incrementing
- the user can leave a comment by clicking the comment button and then provide a comment then submit it
- for each comment, the comment section counter increments
- A demonstration video on how to make the recipe provided in the popup

https://www.loom.com/share/4630e8631e104476b5172cf962a3177c

### CODE REVIEW

File: src/modules/displayMeals.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import fetchMeals from './fetchMeals.js';
+import renderMeals from './renderMeals.js';
+import selectCard from './popup.js';
+import { getLikesNumber, fetchLikes } from './likes.js';
+
+const involvementUrl = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/CUZJYbcXhLNNl7JTkeKF/likes';
+
+const displayMeals = async (mealUrl) => {
+  console.log(mealUrl);
```
Comment (aimemalaika) [inline src/modules/displayMeals.js:9]: - [ ] [OPTIONAL] Kindly remove console logs in your deployed codes

File: src/modules/likes.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+  likesNumber = data[id].likes;
+  return likesNumber;
+};
+
+export const fetchLikes = async (url) => {
+  try {
+    const response = await fetch(url);
+    const data = await response.json();
+    // likesCounter(data);
```
Comment (aimemalaika) [inline src/modules/likes.js:16]: - [ ] [OPTIONAL] Kindly remove all commented code as it makes your repository clean and readable.

File: src/modules/searchform.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+const form = document.querySelector('.search-form');
+const mealInput = document.querySelector('.meal-input');
+
+form.addEventListener('keyup', (e) => {
+  e.preventDefault();
+  console.log('hello');
+
+  const { value } = mealInput;
+  console.log(value);
```
Comment (aimemalaika) [inline src/modules/searchform.js:13]: - [ ][OPTIONAL] Kindly avoid pushing code containing the console log.

**Review by aimemalaika — CHANGES_REQUESTED** [review]
Hi @MohammedElseddik & @evans22j, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

the-simpsons-homer-simpson


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ] Kindly add recorded a video presenting the features of the project.
> The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/comment.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+const data = [{ comment: 'amazing', userName: 'Evans', date: '12/12/2112' }, { comment: 'i LOVE IT', userName: 'CRIS', data: '22/02/2020' }];
+const commentItem = document.querySelector('.allComments');
+data.forEach((elem) => {
+  const newComment = document.createElement('li');
+  newComment.innerHTML += `${elem.creation_date} ${elem.username}: ${elem.comment}`;
+  commentItem.appendChild(newComment);
+});
+console.log([commentItem.children].length);
+
```
Comment (ahmetbozaci) [inline src/__test__/comment.test.js:21]: ### OPTIONAL

_**I write it as an optional but this is a requirement that is not mentioned in the last review. So please fix it.**_

- [ ] You should delete the console.log as last TSE said. Last TSE said optional but this is a [requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md#:~:text=Do%20not%20commit%20console.log%20to%20your%20repo.%20It%E2%80%99s%20ugly%2C%20it%20kills%20performance%20and%20it%20can%20make%20confidential%20data%20be%20visible%20to%20anyone%20using%20the%20browser%20tools%20to%20look%20at%20your%20website.).

File: src/modules/searchform.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+const URL = 'https://www.themealdb.com/api/json/v1/1/search.php?s=';
+
+const form = document.querySelector('.search-form');
+const mealInput = document.querySelector('.meal-input');
+
+form.addEventListener('keyup', (e) => {
+  e.preventDefault();
+  const { value } = mealInput;
+  console.log(value);
```
Comment (ahmetbozaci) [inline src/modules/searchform.js:11]: ### OPTIONAL

_**I write it as an optional but this is a requirement that is not mentioned in the last review. So please fix it.**_

- [ ] You should delete the console.log as last TSE said. Last TSE said optional but this is a [requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md#:~:text=Do%20not%20commit%20console.log%20to%20your%20repo.%20It%E2%80%99s%20ugly%2C%20it%20kills%20performance%20and%20it%20can%20make%20confidential%20data%20be%20visible%20to%20anyone%20using%20the%20browser%20tools%20to%20look%20at%20your%20website.).

**Review by ahmetbozaci — APPROVED** [review]
# STATUS: APPROVED 🥳 🥳 🥳 



@MohammedElseddik @evans22j  ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.

- Nice to add a presentation video. It is clear and professional. But video length should be [5 minutes max.](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md#:~:text=.%20It%20is%20a%20video%20with%20a%20maximum%20length%20of%205%20minutes.%20The%20content%20of%20the%20video%20should%20include%3A) Please so you should pay attention to this in your next projects.

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

aproved

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**